x-pack/plugins/security_solution/public/flyout/document_details/right/components/graph_preview_container.tsx

```diff
@@ -54,7 +71,38 @@ export const GraphPreviewContainer: React.FC = () => {
               defaultMessage="Graph preview"
             />
           ),
-          iconType: 'indexMapping',
+          headerContent: (
+            <EuiBetaBadge
+              alignment="middle"
+              iconType="beaker"
+              data-test-subj="graphPreviewBetaBadge"
+              label={i18n.translate(
+                'xpack.securitySolution.flyout.right.visualizations.graphPreview.technicalPreviewLabel',
+                {
+                  defaultMessage: 'Technical Preview',
+                }
+              )}
+              tooltipContent={i18n.translate(
+                'xpack.securitySolution.flyout.right.visualizations.graphPreview.technicalPreviewTooltip',
+                {
+                  defaultMessage:
+                    'This functionality is in technical preview and may be changed or removed completely in a future release. Elastic will work to fix any issues, but features in technical preview are not subject to the support SLA of official GA features.',
+                }
+              )}
+            />
+          ),
+          iconType: visualizationInFlyoutEnabled ? 'arrowStart' : 'indexMapping',
```

Comment:
similar to @PhilippeOberti 's point above, `visualizationInFlyoutEnabled` is checked in the parent component so it will always be true here.

I do think we should check `isPreviewMode` and `isPreivew` here, when an alert is opened in preview panel or in rule preview, we currently disabled the ability to open a left panel (otherwise the left and right share different alert context)
You may have to do a similar check like in [here](https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/public/flyout/document_details/right/components/analyzer_preview_container.tsx#L81-L83)
<img width="967" alt="image" src="https://github.com/user-attachments/assets/b8175603-136d-4402-9a7f-da21cc8625da">

Reply:
thank you @christineweng , I added the checks